clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1146,9 +1164,19 @@ public InFlightRequest(RequestHeader header,
             this.sendTimeMs = sendTimeMs;
         }
 
+        public ClientResponse completed(AbstractResponse response, long timeMs, MemoryPool memoryPool,
+            ByteBuffer responsePayload, LogContext logContext, boolean enableClientResponseWithFinalize) {
+            if (enableClientResponseWithFinalize) {
+                return new ClientResponseWithFinalize(header, callback, destination, createdTimeMs, timeMs, false, null, null, response,
+                    memoryPool, responsePayload, logContext);
+            }
+            return new ClientResponse(header, callback, destination, createdTimeMs, timeMs, false, null, null, response,
```

Comment:
This should call the second existing constructor and not the new one, right?

Reply:
Not sure if I grok you completely. I think it needs to call this constructor as the request got a successful response and some memory was allocated so we need to pass in the response ByteBuffer and MemoryPool return it back to once the fetch request is done with all completedFetch objects. Let me know if I butchered it